### DISCUSSION TITLE
source .env not working

### DISCUSSION THREAD

**Discussion by curlybeard** [Cyfrin/foundry-full-course-cu#107] [Q&A]
Hello friends,

I am using ubuntu via WSL on windows and am running into the following error when sourcing my .env file. I have asked our AI buddies, and they suggested the presence of invisible characters, but I've recreated the .env file multiple times, ensured there are no spaces and even typed in manually the initialization value.

sample .env file:

```
SEPOLIA_RPC_URL=alchemyApiKey
```

output:
```
source .env
: command not foundb08f-08aba34c9257
: command not found
: command not found
```

Thank you for your time

**Comment by alymurtazamemon** [comment]
@hillowner45 Check out this [link](https://stackoverflow.com/questions/13702425/source-command-not-found-in-sh-shell)

**Reply by cromewar** [reply]
Did @alymurtazamemon solution worked for you @hillowner45?

**Reply by curlybeard** [reply]
@cromewar No unfortunately, as I am not missing the commands to source the .env file.

I've tried

```
source .env
```

and 

```
. .env
```

It seems the issue is with the content of the .env file itself, but im quite certain i've formatted the declaration and initialization of the variable correctly, with no spaces between the '=' and the declared and initialized variable/value.

**Reply by alymurtazamemon** [reply]
@hillowner45 Can we see the content? Hide your keys!

**Reply by curlybeard** [reply]
@alymurtazamemon Thanks for the remainder :D 

All my .env file contains is this 1 variable

```
SEPOLIA_PRC_URL=https://eth-sepolia.g.alchemy.com/v2/hexidecimalString
```

And I just tried sourcing an empty .env file, and it still gives the same 'command not found' error as above.

**Reply by curlybeard** [reply]
Just to show evidence, that I am in the bash shell and that the source commands are installed:

```
root@DESKTOP-F01:~/Foundy-f23/foundry-fund-me-f23# type source
source is a shell builtin
root@DESKTOP-F01~/Foundy-f23/foundry-fund-me-f23# type .
. is a shell builtin
root@DESKTOP-F01:~/Foundy-f23/foundry-fund-me-f23# echo $0
/bin/bash
root@DESKTOP-F01~/Foundy-f23/foundry-fund-me-f23# source .env
: command not foundb08f-08aba34c9257
: command not found
: command not found
```

**Reply by PatrickAlphaC** [reply]
run this

```
dos2unix .env
```

And see if that works? The `.env` file could be a windows .env file and needs to be "linuxed"

**Reply by curlybeard** [reply]
Still no luck after converting the .env file to linux/unix style unfortunately 

```
root@DESKTOP-F01:~/Foundy-f23/foundry-fund-me-f23# dos2unix .env
dos2unix: converting file .env to Unix format...
root@DESKTOP-F01:~/Foundy-f23/foundry-fund-me-f23# source .env
: command not foundb08f-08aba34c9257
: command not found
: command not found
```

**Comment by npa528** [comment]
Try 

```
 sudo source .env
```

Also use 
```
ls -la
```
in your directory where .env is, to check its permissions.

**Reply by PatrickAlphaC** [reply]
Also, have you tried working with chatgpt on this?

**Comment by nehal2210** [comment]
@hillowner45  you got three times "command not found "errors so I am supposing you have more than one field in your .env file.
I was getting the same error, it was solved by removing the space between variable, value and =.
for example:
before  -> PRIVATE_KEY = 0x##### (error)
after -> PRIVATE_KEY=0x##### (solution)

yaa I know this is weird, but it works for me.

**Reply by GauravB2004** [reply]
I had a similar issue in sourcing the env file. Also, I was getting the error that rpc URL and private key not found. But after removing the spaces in the env file, it worked for me. Thanks!

**Reply by yash-2138** [reply]
This worked for me.  Thanks !

**Reply by 0xCardiE** [reply]
yes, had the same problem with white spaces
private_key = "344343"  doesn't work and works only if I make
private_key="344343"

**Reply by gulshanpr** [reply]
hey, @nehal2210  Thanks, I had the similar issue and after removing the spaces it worked for me.

**Comment by 0xTacoTruck** [comment]
@nehal2210 - This is what solved the issue for me also. Awesome!

**Comment by BondeYash** [comment]
@nehal2210 Thanks man This Works for me 
after removing spaces the source .env command works

**Reply by mehedihassan0** [reply]
how and where do u get the private key files to remove space? i am stuck with same .env not found error

**Reply by shikhar229169** [reply]
So, if you are writing something like below it will not work because there are spaces on left and right side of '=' symbol
```env
RPC_URL = http://127.0.0.1:8545
```

So, you need to refactor it like this:
```env
RPC_URL=http://127.0.0.1:8545
```
